src/batArray.mlv

```diff
@@ -823,12 +823,15 @@ let insert xs x i =
 *)
 
 let remove_at i src =
-  let x = src.(i) in (* keep the bound check in there *)
   let n = length src in
-  let dst = make (n - 1) x in
-  blit src 0 dst 0 i;
-  blit src (i + 1) dst i (n - i - 1);
-  dst
+  if i = 0 then
+    sub src 1 (n - 1)
+  else if i = n - 1 then
+    sub src 0 (n - 1)
+  else
+    let dst = sub src 0 (n - 1) in
```

Comment:
If we `remove_at 1`, most of the array will be copied twice (once in the `sub` and once in the `blit`). Isn't that a problem if the array is large?

Reply:
Note: one way to avoid this would be to add a new function to the standard library `Array` module that exposes the underlying `gather` primitive implemented in C, with a type of the form `('a array * int * int) array -> 'a array`: it takes an array of "subarrays" (array, offset, length), and joins all subarrays into a single output array. This is used under the hood to implement `sub` and `concat`, but it could also be used to implement an efficient `remove_at`.